providers/abstract.go

```diff
@@ -16,6 +18,35 @@ type ConfigProvider interface {
 	GetProviderId() string
 	Get(path string) interface{}
 	GetOrDefault(path string, defaultValue interface{}) interface{}
+	GetInstanceForConsumer(resourceName string) (*BlockInstanceDetails, error)
+	GetInstanceOperator(instanceId string) (*InstanceOperator, error)
+	GetInstancesForProvider(resourceName string) ([]*BlockInstanceDetails, error)
+}
+
+type DefaultCredentials struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+type InstanceOperatorPort struct {
+	Protocol string `json:"protocol"`
+	Port     int    `json:"port"`
+}
+
+type InstanceOperator struct {
+	Hostname    string                          `json:"hostname"`
+	Ports       map[string]InstanceOperatorPort `json:"ports"`
+	Path        string                          `json:"path,omitempty"`
+	Query       string                          `json:"query,omitempty"`
+	Hash        string                          `json:"hash,omitempty"`
+	Credentials any                             `json:"credentials,omitempty"`
```

Comment:
This is not a DefaultCredentials?

Reply:
It's still not - it can be anything so DefaultCredentials is just there as a helper